tldw_Server_API/app/core/DB_Management/Watchlists_DB.py

```
@@ -1430,3 +1430,20 @@ def mark_seen_item(
                     "INSERT INTO source_seen_items (source_id, item_key, etag, last_modified, first_seen_at, last_seen_at) VALUES (?, ?, ?, ?, ?, ?)",
                     (source_id, item_key, etag, last_modified, ts, ts),
                 )
+
+    def list_seen_item_keys(self, source_id: int, *, limit: Optional[int] = None) -> List[str]:
+        sql = "SELECT item_key FROM source_seen_items WHERE source_id = ? ORDER BY last_seen_at DESC"
+        params: Tuple[Any, ...] = (source_id,)
+        if isinstance(limit, int) and limit > 0:
+            try:
+                sql = sql + " LIMIT ?"
+                params = (source_id, int(limit))
+            except Exception:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)